c/tskit/tables.c

```diff
@@ -8105,6 +8105,285 @@ tsk_table_collection_subset(
     return ret;
 }
 
```

Comment:
I like factoring out the 'check shared overlap' function - but, what about doing only that? So, rename `tsk_union_init`  to, say `tsk_check_subset_equality`, and put the `if (check_shared_portion)` inside of `tsk_table_collection_union`? Checking the node map would have to move into `union`, but I think that's it?  This would also be clearer because some of this work (eg making `self_nodes` and `other_nodes` and init'ing self_copy and other_copy) doesn't need to happen if we're not checking the shared portion.

Reply:
agree!